pkg/util/genesis_generator/gen_test.go

```diff
@@ -3,12 +3,92 @@ package genesis_generator
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"github.com/wavesplatform/gowaves/pkg/crypto"
 	"github.com/wavesplatform/gowaves/pkg/proto"
 )
 
 func TestGenerate(t *testing.T) {
-	rs, err := Generate(1558516864282, 'W', proto.MustKeyPair([]byte("test")), uint64(9000000000000000))
+	a, err := proto.MustKeyPair([]byte("test")).Addr('W')
 	require.NoError(t, err)
-	require.Equal(t, 1, rs.TransactionCount)
+	block, err := GenerateGenesisBlock('W', []GenesisTransactionInfo{{Address: a, Amount: 9000000000000000, Timestamp: 1558516864282}}, 153722867, 1558516864282)
+	require.NoError(t, err)
+	require.Equal(t, 1, block.TransactionCount)
+	ok, err := block.VerifySignature('W')
+	require.NoError(t, err)
+	assert.True(t, ok)
+}
+
+func TestGenerateMainNet(t *testing.T) {
+	txs := []GenesisTransactionInfo{
+		{Address: proto.MustAddressFromString("3PAWwWa6GbwcJaFzwqXQN5KQm7H96Y7SHTQ"), Amount: 9999999500000000, Timestamp: 1465742577614},
+		{Address: proto.MustAddressFromString("3P8JdJGYc7vaLu4UXUZc1iRLdzrkGtdCyJM"), Amount: 100000000, Timestamp: 1465742577614},
+		{Address: proto.MustAddressFromString("3PAGPDPqnGkyhcihyjMHe9v36Y4hkAh9yDy"), Amount: 100000000, Timestamp: 1465742577614},
+		{Address: proto.MustAddressFromString("3P9o3ZYwtHkaU1KxsKkFjJqJKS3dLHLC9oF"), Amount: 100000000, Timestamp: 1465742577614},
+		{Address: proto.MustAddressFromString("3PJaDyprvekvPXPuAtxrapacuDJopgJRaU3"), Amount: 100000000, Timestamp: 1465742577614},
+		{Address: proto.MustAddressFromString("3PBWXDFUc86N2EQxKJmW8eFco65xTyMZx6J"), Amount: 100000000, Timestamp: 1465742577614},
+	}
+	sig := crypto.MustSignatureFromBase58("FSH8eAAzZNqnG8xgTZtz5xuLqXySsXgAjmFEC25hXMbEufiGjqWPnGCZFt6gLiVLJny16ipxRNAkkzjjhqTjBE2")
+	block, err := RecreateGenesisBlock('W', txs, 153722867, 1460678400000, sig)
+	require.NoError(t, err)
+	bb, err := block.MarshalBinary()
+	require.NoError(t, err)
+	assert.Equal(t, 500, len(bb))
+	assert.Equal(t, 6, block.TransactionCount)
+	assert.Equal(t, 283, int(block.TransactionBlockLength))
+	assert.ElementsMatch(t, sig.Bytes(), block.BlockSignature.Bytes())
```

Comment:
Better use `Equal` instead of `ElementsMatch`. When you use `ElementsMatch` you have no guarantees with elements order. Correct this in all places.

Reply:
Fixed